### PR TITLE
Feature - Azure - Enabling PoP Token verification #331

### PR DESCRIPTION
**Pull request by julienstroheker**
This PR allows any Azure Auth provider to leverage AAD based authentication using tokens with [proof of possession (PoP)](https://datatracker.ietf.org/doc/html/rfc7800).

Introducing :
- New flag `azure.enable-pop` to either activate or not this verification
- New flag `azure.pop-hostname` used by PoP to validate the `u` claim been passed
- New flag `azure.pop-token-validity-duration` to set a TTL duration for the PoP token been passed
- New flag `azure.skip-group-membership-resolution` used to bypass the group membership resolution logic
- New `PassthroughAuthMode` used when we don't want to run any extra validation against GRAPH such as refreshing or get a new token. This scenario is mainly used by Azure Arc today since `AKSAuthMode` `OBOAuthMode` `ClientCredentialAuthMode` cannot been used
- New `PoPTokenVerifier` Struct used to verify the PoP token structure. As today this is non official and custom code but following up with AAD Team to see how and when this verification will be official and been supported in GO-MSAL ?
- Adding UTs for each new components

### CODE REVIEW

File: auth/providers/azure/pop_tokenverifier.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		}
+	} else {
+		return "", errors.Errorf("Invalid token. 'ts' claim is missing")
+	}
+
+	// Verify host 'u' claim
+	if uc, ok := claims["u"]; ok {
+		if reqHostName, ok := uc.(string); ok {
+			if klog.V(10).Enabled() {
+				klog.V(10).Infoln("pop token validation running with hostName: %s. Request is coming for hostName: %s", p.hostName, reqHostName)
```
[the remark this answers is not in the document]
Comment (krdhruva) [reply auth/providers/azure/pop_tokenverifier.go]: What is default level you are setting. Usually it's 5. If you want this statement only during debug time then 6/7 would work.

**Review by krdhruva — APPROVED** [review]

**Comment by tamalsaha** [comment]
@julienstroheker , could you please fix the test?

File: auth/providers/azure/pop_tokenverifier.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (julienstroheker) [reply auth/providers/azure/pop_tokenverifier.go]: I see. Thanks

**Comment by julienstroheker** [comment]
@tamalsaha should be good now, Not sure how to get rid of the `DCO` gate though. I did run the command

**Review by weinong — APPROVED** [review]

**Comment by tamalsaha** [comment]
Tagged https://github.com/kubeguard/guard/releases/tag/v0.9.0

**Pull request merged**